src/query/storages/fuse/src/table_functions/fuse_encodings/fuse_encoding.rs

```diff
@@ -17,151 +17,213 @@ use std::sync::Arc;
+    pub tables: Vec<&'a FuseTable>,
     pub limit: Option<usize>,
+    pub filters: Option<Filters>,
 }
 
 impl<'a> FuseEncoding<'a> {
     pub fn new(
         ctx: Arc<dyn TableContext>,
-        table: &'a FuseTable,
-        column: String,
+        tables: Vec<&'a FuseTable>,
         limit: Option<usize>,
+        filters: Option<Filters>,
     ) -> Self {
         Self {
             ctx,
-            table,
-            column,
+            tables,
             limit,
+            filters,
         }
     }
 
     #[async_backtrace::framed]
     pub async fn get_blocks(&self) -> Result<DataBlock> {
-        let snapshot = self.table.read_table_snapshot().await?;
-        if snapshot.is_none() {
-            return Ok(DataBlock::empty_with_schema(Arc::new(
-                Self::schema().into(),
-            )));
-        }
-        let snapshot = snapshot.unwrap();
-
-        let segments_io = SegmentsIO::create(
-            self.ctx.clone(),
-            self.table.operator.clone(),
-            self.table.schema(),
-        );
-
-        let chunk_size = self.ctx.get_settings().get_max_threads()? as usize * 4;
-
-        let schema = self.table.schema();
-        let field = schema.field_with_name(&self.column)?;
-        if field.is_nested() {
-            return Err(ErrorCode::Unimplemented(format!(
-                "Nested column {} is not supported yet",
-                self.column
-            )));
-        }
-        let column_id = field.column_id;
-        let arrow_field: Field = field.into();
-        let mut pages_info = vec![];
-        for chunk in snapshot.segments.chunks(chunk_size) {
-            let segments = segments_io
-                .read_segments::<SegmentInfo>(chunk, false)
-                .await?;
-            for segment in segments {
-                let segment = segment?;
-                for block in segment.blocks.iter() {
-                    let column_meta = block.col_metas.get(&column_id).unwrap();
-                    let (offset, len) = column_meta.offset_length();
-                    let ranges = vec![(column_id, offset..(offset + len))];
-                    let read_settings = ReadSettings::from_ctx(&self.ctx)?;
-                    let merge_io_read_res = BlockReader::merge_io_read(
-                        &read_settings,
-                        self.table.operator.clone(),
-                        &block.location.0,
-                        ranges,
-                        true,
-                    )
+        let mut info = Vec::new();
+        for table in self.tables.clone() {
+            if matches!(table.storage_format, FuseStorageFormat::Parquet) {
+                continue;
+            }
+            let mut columns_info = vec![];
+            let snapshot = table.read_table_snapshot().await?;
+            if snapshot.is_none() {
+                return Ok(DataBlock::empty_with_schema(Arc::new(
+                    Self::schema().into(),
+                )));
```

Comment:
It seems that a table with no snapshots will end up stopping the collection of info from all the other tables that come after it, which is not expected

Reply:
ok, I will fix it